src/clm/commands/write_nn_Tc.py

```diff
@@ -24,29 +24,46 @@ def calculate_fingerprint(smile):
     return None
 
 
-def find_max_similarity_fingerprint(target_smile, ref_smiles, ref_fps):
-    target_fps = calculate_fingerprint(target_smile)
+def find_max_similarity_fingerprint(target_row, ref_smiles, ref_fps, ref_inchikeys):
+    target_fps = calculate_fingerprint(target_row.smiles)
+    try:
+        target_inchikey = target_row.inchikey
+    except AttributeError:
+        target_inchikey = target_row.target_inchikey
 
     if target_fps is None:
         return None, None
 
-    tcs = [FingerprintSimilarity(target_fps, ref_fp) for ref_fp in ref_fps]
+    tcs = []
+    for ref_smile, ref_fp, ref_inchikey in zip(ref_smiles, ref_fps, ref_inchikeys):
+        # Avoid comparing tcs of exactly same molecule
+        if not (target_inchikey == ref_inchikey):
+            tcs.append(FingerprintSimilarity(target_fps, ref_fp))
+        else:
+            tcs.append(-1)
 
     return np.max(tcs), ref_smiles[np.argmax(tcs)]
 
 
 def write_nn_Tc(query_file, reference_file, output_file):
-    ref_fps, ref_smiles = [], []
-    for smile in read_file(reference_file, stream=True, smile_only=True):
-        if (fps := calculate_fingerprint(smile)) is not None:
+    ref_fps, ref_smiles, ref_inchikeys = [], [], []
+    for row in pd.read_csv(reference_file, chunksize=1, iterator=True):
```

Comment:
To avoid the repeated use of `[0]` in the loop, let's have a `row = df.iloc[0]` and then get whatever you wants out of the `row`. (call this original `row` variable `df` and put a comment on why we're doing all this - for memory saving etc).